index.js

```diff
@@ -1,61 +1,188 @@
+'use strict';
+
 const core = require('@actions/core');
 const github = require('@actions/github');
 const http = require('@actions/http-client');
 
-async function main() {
-    let sha = github.context.sha;
+class GitlabClient {
+    constructor (inputConfig, githubConfig) {
+        this.inputConfig = inputConfig;
+        this.githubConfig = githubConfig;
+        this.client = new http.HttpClient();
+    }
+
+    async request (method, url, data) {
```

Comment:
Just a nitpick:

I see that there are `get` and `post` methods, in JS it's worth annotating private functions with an underscore to show clear intent that it's not supposed to be callable from the outside like `async _request(...)`.